modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/Common.cs

```diff
@@ -358,5 +358,31 @@ ISymbol exportedMemberSymbol
                 typeArgumentSyntax.GetLocation(),
                 typeArgumentSyntax.SyntaxTree.FilePath));
         }
+
+        static SymbolDisplayFormat _symbolDisplayFormat = new SymbolDisplayFormat(typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameAndContainingTypesAndNamespaces);
+        public static string SyntaxToQualifiedString(SyntaxNode node, SemanticModel model, bool isFirstNode = true)
+        {
+            if (node is IdentifierNameSyntax ns && isFirstNode)
+            {
+                SymbolInfo nameInfo = model.GetSymbolInfo(ns);
+                if (nameInfo.Symbol is null) return ns.ToString();
+                return nameInfo.Symbol.ToDisplayString(_symbolDisplayFormat);
+            }
+
+            string s = "";
+            bool innerIsFirstNode = true;
+            foreach (var child in node.ChildNodesAndTokens())
+            {
+                if (child.AsNode() is SyntaxNode childNode)
```

Comment:
I can do this, but then the compile can't understand that AsNode won't be null. I slightly prefer this since it doesn't trigger any warnings, but I can change it if you'd prefer the other way.

Reply:
I see, I suppose we could use the null forgiveness operator (`!`).